Adds basic atom feed + atom/xml link

donatj opened the pull request:
Adds a basic atom feed for the blog and a global `<link>`

I do believe this solves #251

It produces a syntactically valid ATOM feed 

![image](https://user-images.githubusercontent.com/133747/124699847-3d107b00-deb1-11eb-8253-592e4a7f5d07.png)

donatj commented:
You can now see the feed in question here:

https://www---pr-293-lktmcji-ovkfaqrdltr5s.us.platform.sh/atom.xml

You can see it validated here.

https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fwww---pr-293-lktmcji-ovkfaqrdltr5s.us.platform.sh%2Fatom.xml

A couple minor warnings but no errors.

> Your feed appears to be encoded as "utf-8", but your server is reporting "US-ASCII" 

I think that's just a platform.sh problem

> Self reference doesn't match document location

It will when it's live, site.host just isn't being set right now for platform.sh

> Invalid HTML: Named entity expected. Got none.

It's not a fan of the HTML of the POST. Non-issue really

> content should not contain script tag

It's not really a problem. Some RSS readers will execute it, some won't. You could probably write a filter to scrape out script tags, but it's not really necessary.

Jean85 commented on source/_layouts/base.twig:
```diff
@@ -24,6 +24,8 @@
     <link rel="preconnect" href="https://ajax.googleapis.com" pr="1.0" />
     <link rel="preconnect" href="https://ssl.google-analytics.com" pr="0.8" />
 
+    <link rel="alternate" type="application/atom+xml" href="{{ site.host }}/atom.xml" title="PHP-FIG Blog Feed" />
```
Is `site.host` really needed? All other links are relative, starting with `/` to explicitly point to the root.

donatj replied:
Nope, removed.

Jean85 commented on source/atom.xml:
```diff
@@ -0,0 +1,35 @@
+            <link href="{{ site.host }}{{ post.url }}"/>
+            <updated>{{ post.date|date('c') }}</updated>
+            <id>{{ site.host }}{{ post.url }}</id>
+            <content type="html"><![CDATA[{{ post.blocks.content|raw }}]]></content>
```
Couldn't you just use `striptags` instead of `raw` to cleanup all the post data? https://twig.symfony.com/doc/2.x/filters/striptags.html

donatj replied:
Comments below

Jean85 reviewed (changes requested):
Thank you for the contribution! Just a couple of suggestions..

donatj commented:
I have dropped the `site.host` from the base.twig. 

I also used `striptags` with a whitelist of allowed tags I came up with by grep / uniq-ing the tags of the posts and then removing `<script>`.

I also added h5 and h6 which although were not used, seemed like they _could be_ in the future.

<details>
<summary>the full grep/uniq results</summary>

```
<a
<blockquote
<br
<code
<em
<h2
<h3
<h4
<iframe
<img
<li
<ol
<p
<pre
<script
<strong
<ul
```  
</details>

Beyond that, I left the `|raw` because encoding within the `<![CDATA[ … ]]>` section is unessessary and akin to _double-encoding_ causing readers without mitigations for bad feeds to display the HTML encoded.

Jean85 commented:
Thank you! Last question: the validation is still complaining, are we missing something? It says it's about `explanation`, but it doesn't make sense... I searched for that word and I find only `If this explanation isn’t enough`... Maybe the `’` is the culprit? Maybe we can ditch the `raw` and `CDATA` altogether?

donatj commented:
After trying a _different_ ATOM validator it appears it just doesn't like iframe - and the complaint was about not recognizing the tag name.

I dropped the iframe from the allowed tag list and the error has gone away.

> Maybe we can ditch the raw and CDATA altogether?

I'm neither an expert on Sculpin nor Twig, but based on [Twig's docs](https://twig.symfony.com/doc/3.x/filters/escape.html)'s lack of mention of XML, it's not implied that the default encoding is _XML safe_. HTML provides a lot more named entities than XML, thus the existence of `ENT_XML1` on [htmlentities](https://www.php.net/manual/en/function.htmlentities.php#refsect1-function.htmlentities-parameters).

I'd also note that the sculpin blog skeleton handles it this way as well - https://github.com/sculpin/sculpin-blog-skeleton/blob/5b1172ac8c4bacd0eb0df3efccfd3bcc25dd7dba/source/atom.xml#L25

Jean85 reviewed (approved):
Great! Thank you very much!
